src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/blocks/BlockPlacer.java

```diff
@@ -146,6 +146,10 @@ private boolean isBlacklisted(Material type) {
     private void placeSlimefunBlock(SlimefunItem sfItem, ItemStack item, Block block, Dispenser dispenser) {
         BlockPlacerPlaceEvent e = new BlockPlacerPlaceEvent(dispenser.getBlock(), item, block);
         Bukkit.getPluginManager().callEvent(e);
+        
+        if(!dispenser.getInventory().getViewers().isEmpty()){
+        	e.setCancelled(true);
```

Comment:
There is no point in calling an Event when it gets cancelled anyway.
The whole point of the Event contract is that **others** get a say in the outcome of the Event, not that their changes get overridden otherwise.

If no block gets placed, then there shouldn't even be an Event called.